examples/source_detection/main.go

```diff
@@ -0,0 +1,54 @@
+package main
+
+import (
+	"encoding/json"
+	"os"
+
+	"github.com/anchore/syft/syft/source"
+)
+
+/*
+ This example demonstrates how to create a source object from a generic string input.
+
+ Example inputs:
+    alpine:3.19                              pull an image from the docker daemon, podman, or the registry (based on what's available)
+    ./my.tar                                 interpret a local archive as an OCI archive, docker save archive, or raw file from disk to catalog
+    docker:yourrepo/yourimage:tag            explicitly use the Docker daemon
+    podman:yourrepo/yourimage:tag            explicitly use the Podman daemon
+    registry:yourrepo/yourimage:tag          pull image directly from a registry (no container runtime required)
+    docker-archive:path/to/yourimage.tar     use a tarball from disk for archives created from "docker save"
+    oci-archive:path/to/yourimage.tar        use a tarball from disk for OCI archives (from Skopeo or otherwise)
+    oci-dir:path/to/yourimage                read directly from a path on disk for OCI layout directories (from Skopeo or otherwise)
+    singularity:path/to/yourimage.sif        read directly from a Singularity Image Format (SIF) container on disk
+    dir:path/to/yourproject                  read directly from a path on disk (any directory)
+    file:path/to/yourproject/file            read directly from a path on disk (any single file)
+
+*/
+
+const imageRef = "alpine:3.19"
```

Comment:
I think this could use a different name. Maybe even read from argv, so that it's easier to play with and see what different strings result in.

Reply:
I want folks to be able to run them without needing to know what to pass, but I can change these to respond to argv  with a default if nothing is passed.